libs/ui-toolkit/src/components/async-renderer/async-renderer.tsx

```diff
@@ -25,6 +27,6 @@ export function AsyncRenderer<T = object>({
   if (!data) {
     return <Splash>No data</Splash>;
   }
-
-  return <>{children(data)}</>;
+  // eslint-disable-next-line react/jsx-no-useless-fragment
+  return <>{render ? render(data) : children}</>;
```

Comment:
instead of 
`<AsyncRenderer data={data}>{(data) => <Child data={data}/>}<AsyncRenderer/>`
We can now use
`<AsyncRenderer data={data}render={(data) => <Child data={data} />}/>`
or 
`<AsyncRenderer  data={data}><Child data={data}/></AsyncRenderer >`

Reply:
I mentioned I another comment but this thread is a better spot for it... 

I quite like enforcing to use at least a render prop (I don't mind if its `render` or `children`). This way in the body of the render prop function TS can infer that data is indeed truthy. Otherwise we end up writing pointless null checks
```
// TS cant infer data is truthy
<AsyncRenderer ... >
  {data && <Child />
</AsyncRenderer>

// TS can infer this though
<AsyncRenderer ... 
  render={(data) => (
    <Child data={data} />
  )
/>
```

Reply:
I know what you mean, but TS will not infer if data is null also when it will be used in render function
![image](https://user-images.githubusercontent.com/1754247/164467972-1a67a2be-bfac-4eb3-a3a6-68072f67816b.png)

Only way to make it look nice is to allow <Child /> to accept nullish data, and then we can use both render function or child notation

Reply:
It will but you have to pass AsyncRenderer a generic type argument. Regardless I dont think this is important enough to block the PR 😄